src/Properties/ModelPropertyExtension.php

```diff
@@ -153,6 +153,22 @@ private function getDateClass(): string
         return $this->dateClass;
     }
 
+    /**
+     * @param  Model  $modelInstance
+     * @return string[]
+     * @phpstan-return array<int, string>
+     */
+    private function getModelDateColumns(Model $modelInstance): array
+    {
+        $dateColumns = $modelInstance->getDates();
+
+        if (! method_exists($modelInstance, 'getDeletedAtColumn')) {
```

Comment:
I'd prefer doing a `method_exists` check here instead of `not`. So we can just add the `$modelInstance->getDeletedAtColumn()` to the `$dateColumns` array. This way we can skip array unpacking and returning new array.

Reply:
No problem, that's how I originally wrote it and then thought an early return may be preferable. I'll get it changed now.